### ISSUE TITLE
Twitter bots dataset fails to load most images because 4 channels not supported for JPEG.

### ISSUE THREAD

**Issue by dantreiman** [ludwig-ai/ludwig#1940]
Training with default settings for image feature, image preprocessing samples the first 100 images and takes the max of the number of channels.

image_feature.py:
```python
    if preprocessing_parameters[INFER_IMAGE_DIMENSIONS]:
        user_specified_num_channels = True
        # Use the maximum num_channels found across all sampled images. torchvision has built-in support for
        # upsampling images.
        num_channels = max(num_channels_in_image(x) for x in inferred_sample)
```

This results in `num_channels = 4`, but the majority of images in the dataset are JPEGs, and torchvision's `read_image(img, mode=ImageReadMode.RGB_ALPHA)` throws an exception `The provided mode is not supported for JPEG files`:

```
reading image url /Users/daniel/Downloads/twitter_bots_kaggle/profile_images/ef985bc5d304e6fbe8c1bb65bead9ca24de7bb2c.jpg failed due to The provided mode is not supported for JPEG files
/Users/daniel/Downloads/twitter_bots_kaggle/profile_images/ef985bc5d304e6fbe8c1bb65bead9ca24de7bb2c.jpg cannot be read
reading image url /Users/daniel/Downloads/twitter_bots_kaggle/profile_images/c55fc16d98950aad439bfa44ac68712f1f10accf.jpeg failed due to The provided mode is not supported for JPEG files
....
```

I'm not sure what the right heuristic is here, but I'm pretty certain we shouldn't use alpha channel if the majority of the images in dataset do not have an alpha channel.

**Comment by dantreiman** [comment]
Setting `num_channels` explicitly seems to be a valid workaround:

``` yaml
  - name: profile_image_url
     type: image
     preprocessing:
       num_channels: 3
```

**Comment by dantreiman** [comment]
Fixed by #1941
